src/sentry/issues/endpoints/organization_group_search_views.py

```diff
@@ -96,11 +96,16 @@ def get(self, request: Request, organization: Organization) -> Response:
             )
 
         if not has_global_views:
-            for view in query:
-                if view.is_all_projects or view.projects.count() > 1 or view.projects.count() == 0:
+            views_to_updates = []
+            default_project = pick_default_project(organization, request.user)
+
+            for view in query.prefetch_related("projects"):
+                if view.is_all_projects or view.projects.count() != 1:
                     view.is_all_projects = False
-                    view.projects.set([pick_default_project(organization, request.user)])
-                    view.save()
+                    view.projects.set([default_project])
```

Comment:
Won't happen very often, but I think it would be better if (in the case of multiple selected projects) this chose one of those